torch/csrc/jit/import.cpp

```diff
@@ -131,11 +125,9 @@ template<>
 struct Reader<int64_t> : ReaderBase {
   Reader(pb_callback_t& cb) : ReaderBase(cb) {}
   virtual void decode(pb_istream_t *stream) override {
-    uint64_t val;
-    if (!pb_decode_varint(stream, &val)) {
+    if (!pb_decode_varint(stream, reinterpret_cast<uint64_t*>(&value))) {
```

Comment:
Oh I didn't notice this one isn't the same type! This is really worrying because this means that it's effectively impossible to get a negative `int64_t` from protobuf. Are you sure there's really no other function that would give us a signed number?

Reply:
Yeah, it looks fishy, but apparently this is the way it's intended to be used
(https://github.com/nanopb/nanopb/blob/master/pb_decode.c#L1289 and https://github.com/nanopb/nanopb/blob/master/tests/decode_unittests/decode_unittests.c).
Encoding is similar, see @ezyang's comment (https://github.com/pytorch/pytorch/blob/master/torch/csrc/onnx/onnx.cpp#L12).

I'm obviously not an expert, but from what I read protobuf encodes all integers (signed and unsigned) as a `varint` internally (https://developers.google.com/protocol-buffers/docs/encoding?csw=1#varints). A `varint` is decoded by nanopb into 64 bits, meant to be interpreted as a signed or unsigned integer depending what was originally encoded in the `varint`. Thus negative `int64_t` will be returned as such after the `reinterpret_cast`.